pyccel/parser/semantic.py

```diff
@@ -1357,15 +1357,10 @@ def _handle_function(self, expr, func, args, *, is_method = False, use_build_fun
             return new_expr
         else:
             is_inline = func.is_inline if isinstance(func, FunctionDef) else func.functions[0].is_inline
-            if not func.is_semantic:
+            if is_inline:
+                return self._visit_InlineFunctionDef(func, args, expr)
```

Comment:
Can you run this function on `func` if it is an `Interface` node?

The docstring of `_visit_InlineFunctionDef` says that `func` is of type `InlineFunctionDef`...

Reply:
No. `Interface`s are never created in the syntactic stage. They are used to hold multiple semantic `FunctionDef` objects which have the same name in Python. For inline functions we never create a semantic object so there is no risk of an `Interface` appearing

Reply:
Fixed with 923d665